src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -178,7 +178,8 @@ private IEnumerable<string> GetPluginFilePaths()
         {
             if (string.IsNullOrEmpty(_rawPluginPaths))
             {
-                return Enumerable.Empty<string>();
+                var directories = new List<string> { PluginDiscoveryUtility.GetNuGetHomePluginsPath(), PluginDiscoveryUtility.GetInternalPlugins() };
```

Comment:
`PluginDiscoveryUtility.GetInternalPlugins()` could return nulll which will just be added in `directories` list which will further be processed in next API. You would need to better handle nulls there.